frontend/src/components/task/task-status-icons.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { FormattedMessage } from 'react-intl'
+import {
+  Redeem as RedeemIcon,
+  Lock as PrivateIcon,
+  LockOpen as PublicIcon
+} from '@material-ui/icons'
+import {
+  Paper,
+  Typography
+} from '@material-ui/core'
+
+const styles = {
+  IconStyle: {
+    fontSize: 48
+  },
+  fontStyle: {
+    fontWeight: '900',
+    fontSize: 20,
+    marginBottom: 30
+  }
+}
+
+/* Expects two props
+status: String<'public' | 'private'> (You can pass a dynamic property that evaluates
+                                      to 'public' or 'private')
+bounty: Boolean<true | false>
+*/
+
+const TaskStatusIcons = ({ status, bounty }) => {
+  const Bounty = () => {
+    if (bounty) {
+      return (
+        <React.Fragment>
+          <RedeemIcon style={ styles.IconStyle } />
+          <Typography
+            style={ styles.fontStyle }>
+            <FormattedMessage id='task.bounty.icon.text' defaultMessage='Bounty' />
+          </Typography>
+        </React.Fragment>
+      )
+    }
+    else return <div />
+  }
+  const Status = () => {
+    if (status) {
+      return (
+        <React.Fragment>
+          { (status === 'private')
+            ? (<PrivateIcon style={ styles.IconStyle } />)
+            : (<PublicIcon style={ styles.IconStyle } />)
+          }
+          <Typography
+            style={ styles.fontStyle }>
+            { (status === 'private') ? <FormattedMessage id='task.private.icon.text' defaultMessage='Private' />
+              : <FormattedMessage id='task.public.icon.text' defaultMessage='Public' />
```

Comment:
We can't have same ids, they need different names like `task.private.icon.text.public` and `task.private.icon.text.private`, something like that.

Reply:
They are different only. task.**private**.icon.text and task.**public**.icon.text.

Am I correct ? Or maybe I am not getting what you mean to say.  You can maybe write the exact code.